File: binary_search/binary_search.go
package main

import "fmt"

func binarySearch(list []int, item int) int {
    low := 0
    high := len(list) - 1

    for low <= high {
        mid := (low + high) / 2

        if list[mid] == item {
            return mid
        }

        if list[mid] < item {
            low = mid + 1
        } else {
            high = mid - 1
        }
    }
    return -1
}

func main() {
    fmt.Println(binarySearch([]int{1, 3, 5, 7, 9}, 5)) // 2
    fmt.Println(binarySearch([]int{1, 3, 5, 7, 9}, -1)) // -1
}

File: binary_search/binary_search.php
<?php

function binary_search($list, $item) {
    $low = 0;
    $high = count($list) - 1;

    while ($low <= $high) {
        $mid = floor(($low + $high) / 2);

        if ($list[$mid] === $item) {
            return $mid;
        }

        if ($list[$mid] > $item) {
            $high = $mid - 1;
        } else {
            $low = $mid + 1;
        }
    }

    return null;
}

$my_list = [1, 3, 5, 7, 9];
echo binary_search($my_list, 3); // 1
echo binary_search($my_list, -1); // null
